tests/test_extract_heads.py

```diff
@@ -0,0 +1,50 @@
+from extract_heads import (
+    most_common_ancestor,
+    normalize_token_to_span,
+    set_extensions,
+)
+import spacy
+from spacy.tokens import Span
+from relationextraction import SpacyRelationExtractor
+from collections import Counter
+
+
+def test_normalize_token():
+
+    nlp = spacy.load("en_core_web_lg")
+    doc = nlp("The quick brown fox named Charles, jumps over the lazy dog.")
+    doc1 = nlp("Mette Frederiksen is the Danish politician.")
+
+    normalized_token = normalize_token_to_span(doc[3])
+
+    assert isinstance(normalized_token, Span)
+    assert normalized_token.text == "fox"
+
+    normalized_token = normalize_token_to_span(doc1[1], normalize_to_entity=True)
+
+    assert isinstance(normalized_token, Span)
+    assert normalized_token.text == "Mette Frederiksen"
+
+
+def test_headword():
+
+    nlp = spacy.load("en_core_web_lg")
+    test_sents = ["Mette Frederiksen is the Danish politician."]
+
+    config = {"confidence_threshold": 2.7, "model_args": {"batch_size": 10}}
+    nlp.add_pipe("relation_extractor", config=config)
+
+    pipe = nlp.pipe(test_sents)
```

Comment:
No need to create a pipe for one document.

```suggestion
    doc = nlp(text)
```